auth0/rules
# Caisson ID Check integration

### Description
Caisson ID Check rule.  Caisson validates US driver's licenses and international passports in real time.

### Testing

- Install the Caisson ID Check rule
- Sign up for an account at [Caisson](https://www.caisson.com)
- Go to your [developer tab](https://www.caisson.com/developer) to fetch your API keys 
- In Auth0, set the following rule configurations
  - CAISSON_PUBLIC_KEY
  - CAISSON_PRIVATE_KEY
  - CAISSON_ON_REGISTRATION
    - "true" (require ID Check on registration)
    - "false" / empty
  - CAISSON_LOGIN_FREQUENCY_DAYS
    - "-1" (always)
    - "0" / empty (never)
    - "[uint]" (min days between ID Checks on login)
  - CAISSON_DEBUG
    - "true" (prints errors to console)
    - "false" / empty

Go through your test app's login flow as usual.  You'll be taken through the ID Check process.  Check for caisson attributes on your user's app_metadata.  You should see: 
  - "idcheck_url" (link to your [Caisson Dashboard](https://www.caisson.com/dashboard) with full ID Check results)
  - "last_check" (timestamp of last ID Check)
  - "status" (Pass / Flagged status for the last ID Check)
  - "count" (number of completed ID Checks) 

### Checklist
